lib/mkfile

```diff
@@ -0,0 +1,229 @@
+	ldap.$O\
+	version.$O\
+	getenv.$O\
+	escape.$O\
+	mprintf.$O\
+	telnet.$O\
+	netrc.$O\
+	getinfo.$O\
+	transfer.$O\
+	strcase.$O\
+	easy.$O\
+	security.$O\
+	curl_fnmatch.$O\
+	fileinfo.$O\
+	ftplistparser.$O\
+	wildcard.$O\
+	krb5.$O\
+	memdebug.$O\
+	http_chunks.$O\
+	strtok.$O\
+	connect.$O\
+	llist.$O\
+	hash.$O\
+	multi.$O\
+	content_encoding.$O\
+	share.$O\
+	http_digest.$O\
+	md4.$O\
+	md5.$O\
+	http_negotiate.$O\
+	inet_pton.$O\
+	strtoofft.$O\
+	strerror.$O\
+	amigaos.$O\
+	hostasyn.$O\
+	hostip4.$O\
+	hostip6.$O\
+	hostsyn.$O\
+	inet_ntop.$O\
+	parsedate.$O\
+	select.$O\
+	tftp.$O\
+	splay.$O\
+	strdup.$O\
+	socks.$O\
+	ssh.$O\
+	ssh-libssh.$O\
+	curl_addrinfo.$O\
+	socks_gssapi.$O\
+	socks_sspi.$O\
+	curl_sspi.$O\
+	slist.$O\
+	nonblock.$O\
+	curl_memrchr.$O\
+	imap.$O\
+	pop3.$O\
+	smtp.$O\
+	pingpong.$O\
+	rtsp.$O\
+	curl_threads.$O\
+	warnless.$O\
+	hmac.$O\
+	curl_rtmp.$O\
+	openldap.$O\
+	curl_gethostname.$O\
+	gopher.$O\
+	idn_win32.$O\
+	http_proxy.$O\
+	non-ascii.$O\
+	asyn-ares.$O\
+	asyn-thread.$O\
+	curl_gssapi.$O\
+	http_ntlm.$O\
+	curl_ntlm_wb.$O\
+	curl_ntlm_core.$O\
+	curl_sasl.$O\
+	rand.$O\
+	curl_multibyte.$O\
+	hostcheck.$O\
+	conncache.$O\
+	pipeline.$O\
+	dotdot.$O\
+	x509asn1.$O\
+	http2.$O\
+	smb.$O\
+	curl_endian.$O\
+	curl_des.$O\
+	system_win32.$O\
+	mime.$O\
+	sha256.$O\
+	setopt.$O\
+	curl_path.$O\
+	curl_ctype.$O\
+	curl_range.$O\
+	psl.$O\
+	doh.$O\
+	urlapi.$O\
+
+HFILES=\
+	arpa_telnet.h\
+	netrc.h\
+	file.h\
+	timeval.h\
+	hostip.h\
+	progress.h\
+	formdata.h\
+	cookie.h\
+	http.h\
+	sendf.h\
+	ftp.h\
+	url.h\
+	dict.h\
+	if2ip.h\
+	speedcheck.h\
+	urldata.h\
+	curl_ldap.h\
+	escape.h\
+	telnet.h\
+	getinfo.h\
+	strcase.h\
+	curl_sec.h\
+	memdebug.h\
+	http_chunks.h\
+	curl_fnmatch.h\
+	wildcard.h\
+	fileinfo.h\
+	ftplistparser.h\
+	strtok.h\
+	connect.h\
+	llist.h\
+	hash.h\
+	content_encoding.h\
+	share.h\
+	curl_md4.h\
+	curl_md5.h\
+	http_digest.h\
+	http_negotiate.h\
+	inet_pton.h\
+	amigaos.h\
+	strtoofft.h\
+	strerror.h\
+	inet_ntop.h\
+	curlx.h\
+	curl_memory.h\
+	curl_setup.h\
+	transfer.h\
+	select.h\
+	easyif.h\
+	multiif.h\
+	parsedate.h\
+	tftp.h\
+	sockaddr.h\
+	splay.h\
+	strdup.h\
+	socks.h\
+	ssh.h\
+	curl_base64.h\
+	curl_addrinfo.h\
+	curl_sspi.h\
+	slist.h\
+	nonblock.h\
+	curl_memrchr.h\
+	imap.h\
+	pop3.h\
+	smtp.h\
+	pingpong.h\
+	rtsp.h\
+	curl_threads.h\
+	warnless.h\
+	curl_hmac.h\
+	curl_rtmp.h\
+	curl_gethostname.h\
+	gopher.h\
+	http_proxy.h\
+	non-ascii.h\
+	asyn.h\
+	http_ntlm.h\
+	curl_gssapi.h\
+	curl_ntlm_wb.h\
+	curl_ntlm_core.h\
+	curl_sasl.h\
+	curl_multibyte.h\
+	hostcheck.h\
+	conncache.h\
+	curl_setup_once.h\
+	multihandle.h\
+	setup-vms.h\
+	pipeline.h\
+	dotdot.h\
+	x509asn1.h\
+	http2.h\
+	sigpipe.h\
+	smb.h\
+	curl_endian.h\
+	curl_des.h\
+	curl_printf.h\
+	system_win32.h\
+	rand.h\
+	mime.h\
+	curl_sha256.h\
+	setopt.h\
+	curl_path.h\
+	curl_ctype.h\
+	curl_range.h\
+	psl.h\
+	doh.h\
+	urlapi-int.h\
+
+LIB=/$objtype/lib/ape/libcurl.a
+</sys/src/cmd/mklib
```

Comment:
I would urge you to either make this use the `lib/Makefile.inc` file directly somehow, or if that isn't possible, make a build script that generates the necessary include file using that file as input. We do change file names and remove/add files every now and then so having the list of source files in separate files like this will make this port rather fragile going forward.

Reply:
ok, I do it.

Reply:
Any update on this?

Reply:
Actually, reading further, this is addressed with the below mkfile.inc?

Reply:
Yes, I updated **mkfile** so that it use **mkfile.inc**.
The operator `<|` on **mkfile** executes an argument, and evals *stdout* of the result.

Reply:
Gotcha, I was commenting a bit prematurely before reading all of the diff.